src/bokeh/models/nodes.py

```diff
@@ -0,0 +1,214 @@
+
+__all__ = (
+    "Indexed",
+    "Node",
+    "XY",
+)
+
+#-----------------------------------------------------------------------------
+# General API
+#-----------------------------------------------------------------------------
+
+ImplicitTarget = Literal["viewport", "canvas", "plot", "frame", "parent"]
+
+class Coordinate(Model):
+    """
+    """
+
+    # explicit __init__ to support Init signatures
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+class XY(Coordinate):
+    """
+    A point in a Cartesian coordinate system.
+
+    .. note::
+        This model is experimental and may change at any point.
+    """
+
+    # explicit __init__ to support Init signatures
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+    x = Required(CoordinateLike, help="""
+    The x component.
+    """)
+
+    y = Required(CoordinateLike, help="""
+    The y component.
+    """)
+
+class Indexed(Coordinate):
+    """
+    """
+
+    # explicit __init__ to support Init signatures
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+    i = Required(Int())
+    renderer = Instance(".models.renderers.GlyphRenderer")
+
+class Node(Coordinate):
+    """
+    Represents a symbolic coordinate (by name).
+
+    .. note::
+        This model is experimental and may change at any point.
+    """
+
+    # explicit __init__ to support Init signatures
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+    target = Required(Either(Instance(Model), Enum(ImplicitTarget)), help="""
+    The provider of coordinates for this node.
+
+    This can be either a concrete model that can provide its coordinates (e.g.
+    a renderer, a frame or a canvas) or an implicit target defined by the
+    enum, which is resolved as the nearest parent of the given type. If the
+    provider cannot be determined or it isn't able to provide coordinates,
+    then the node resolved to an invalid coordinate (with x and y components
+    being ``NaN``).
+    """)
+
+    symbol = Required(String, help="""
+    A symbolic name of a coordinate to provide.
+
+    The allowed terms are dependent on the target of this node. For example,
+    for box-like targets this will comprise of box anchors (e.g. center, top
+    left) and box edges (e.g. top, left).
+    """)
+
+    offset = Int(default=0, help="""
+    Optional pixel offset for the computed coordinate.
+    """)
+
+    @classmethod
+    @property
+    def canvas(cls) -> BoxNodes:
+        return BoxNodes("canvas")
+
+    @classmethod
+    @property
+    def plot(cls) -> BoxNodes:
+        return BoxNodes("plot")
+
+    @classmethod
+    @property
+    def frame(cls) -> BoxNodes:
+        return BoxNodes("frame")
+
+    @classmethod
+    @property
+    def parent(cls) -> BoxNodes:
+        return BoxNodes("parent")
```

Comment:
I have not encountered classmethod properties before. I guess `HasProps` machinery prevents just adding these as class attributes? (Or are new instances really required on every access?)

Reply:
I'm not sure why this is written the way it is, especially given this code already went through multiple superseded PRs. However, I have to change this anyway, because class properties were deprecated in Python 3.11 and will be removed in 3.13.

Reply:
Now I remember. I used class properties to keep the order of definitions in the module.

Reply:
I replaced this class members.